### ISSUE TITLE
`SIM103` should detect implicit `else`

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/ruff#10402]
See: https://github.com/sbdchd/flake8-pie#pie801-prefer-simple-return.

In that example:

```python
# error
def main():
    if foo > 5:
        return True
    return False

# error
def main():
    if foo > 5:
        return True
    else:
        return False

# ok
def main():
    return foo > 5
```

We catch the second case, but not the first.

**Comment by omh1280** [comment]
@charliermarsh I'd like to give this a shot.

**Comment by charliermarsh** [comment]
Great!
